### ISSUE TITLE
Support `concat` alias to `cat`

### ISSUE THREAD

**Issue by AnirudhDagar** [pytorch/pytorch#61767]
## 🚀 Feature
Support `torch.concat`.

## Motivation & Pitch

Python Array API proposes the use of `concat` among other manipulation functions. Currently, PyTorch supports the same functionality with `torch.cat`. As listed in #58742, it would be nice to add an alias `concat` to `cat` making it compliant with Array API.

## Additional context

* [Python Array API concat](https://data-apis.org/array-api/latest/API_specification/manipulation_functions.html#concat-arrays-axis-0)

cc @mruberry @rgommers @pmeier @asmeurer @leofang @AnirudhDagar @asi1024 @emcastillo @kmaehashi @heitorschueroff
